From c6ab08b6734b877f73bf4acf7dcdae4320b6ce9b Mon Sep 17 00:00:00 2001
From: He Zhe <zhe.he@windriver.com>
Date: Tue, 19 Jun 2018 00:58:18 -0700
Subject: [PATCH 2/2] yaffs: Fix build failure by removing unused members of
 erase_info

A few members of struct erase_info have been removed and no longer
used in mainline by the following commits. This patches removes
related useless initialization.

8f347c4232d5fc097599b711a3385722a6834005
"mtd: Unconditionally update ->fail_addr and ->addr in part_erase()"

884cfd9023ce6afe8bcf181ec988d8516eb32bf0
"mtd: Stop assuming mtd_erase() is asynchronous"

dcba51bbb9e0cc7f80d36eb20a033a4dff2ce9cc
"mtd: Get rid of unused fields in struct erase_info"

Signed-off-by: He Zhe <zhe.he@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 fs/yaffs2/yaffs_mtdif.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/fs/yaffs2/yaffs_mtdif.c b/fs/yaffs2/yaffs_mtdif.c
index 03a3d9c24e4c..4ff5741f3983 100644
--- a/fs/yaffs2/yaffs_mtdif.c
+++ b/fs/yaffs2/yaffs_mtdif.c
@@ -54,13 +54,8 @@ int nandmtd_erase_block(struct yaffs_dev *dev, int block_no)
 	struct erase_info ei;
 	int retval = 0;
 
-	ei.mtd = mtd;
 	ei.addr = addr;
 	ei.len = dev->param.total_bytes_per_chunk * dev->param.chunks_per_block;
-	ei.time = 1000;
-	ei.retries = 2;
-	ei.callback = NULL;
-	ei.priv = (u_long) dev;
 
 	retval = mtd_erase(mtd, &ei);
 
@@ -183,13 +178,8 @@ static 	int yaffs_mtd_erase(struct yaffs_dev *dev, int block_no)
 		     dev->param.chunks_per_block;
 	addr = ((loff_t) block_no) * block_size;
 
-	ei.mtd = mtd;
 	ei.addr = addr;
 	ei.len = block_size;
-	ei.time = 1000;
-	ei.retries = 2;
-	ei.callback = NULL;
-	ei.priv = (u_long) dev;
 
 	retval = mtd_erase(mtd, &ei);
 
-- 
2.5.0

